packages/xo-web/src/xo-app/vm/tab-network.js

```diff
@@ -255,37 +261,58 @@ class VifAllowedIps extends BaseComponent {
         )}
         <Row>
           <Col size={10}>
-            {showNewIpForm ? (
-              <span onBlur={this._toggleNewIp}>
+            {showIpSelector && (
+              <span onBlur={this._hideIpInputs}>
                 {resourceSet ? (
                   <SelectResourceSetIp
                     autoFocus
                     containerPredicate={this._getIsNetworkAllowed()}
                     onChange={this._addIp}
                     predicate={this._getIpPredicate()}
                     resourceSetId={resourceSet}
+                    value={null}
                   />
                 ) : (
                   <SelectIp
                     autoFocus
                     containerPredicate={this._getIsNetworkAllowed()}
                     onChange={this._addIp}
                     predicate={this._getIpPredicate()}
+                    value={null}
                   />
                 )}
               </span>
-            ) : (
-              <ActionButton
-                btnStyle='success'
-                size='small'
-                handler={this._toggleNewIp}
-                icon='add'
+            )}
+            {showIpInput && (
+              <input
+                autoFocus
+                onBlur={this._hideIpInputs}
+                onKeyDown={this._onKeyDown}
+                placeholder={formatMessage(messages.addAllowedIp)}
               />
-            )}{' '}
-            {warningMessage !== undefined && (
-              <Tooltip content={warningMessage}>
-                <Icon icon='error' />
-              </Tooltip>
+            )}
+            {!showIpSelector && !showIpInput && (
+              <span>
+                <ActionButton
+                  btnStyle='success'
+                  size='small'
+                  handler={this.toggleState('showIpSelector')}
+                  icon='add'
+                />{' '}
+                {isAdmin && (
+                  <ActionButton
+                    btnStyle='primary'
+                    size='small'
+                    handler={this.toggleState('showIpInput')}
+                    icon='edit'
```

Comment:
IMHO, the edition icon can confuse users because is not an edition of the IP pools but the add of a new IP which doesn't exist in the DB.

So we can add tooltips to the action buttons to make clear their roles or unify the handle of this case by doing like the VM tags' add.